### PR TITLE
[Nursery] Cop-DeepAR: Fix evaluation at all levels

### PR DESCRIPTION
**Pull request by rshyamsundar**
*Issue #, if available:*

Allows the evaluation of time series at all levels of the temporal hierarchy. 

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: src/gluonts/nursery/temporal_hierarchical_forecasting/eval/evaluation.py [hunk 1]
```diff
@@ -87,27 +88,41 @@ def evaluate_forecasts_at_all_levels(
...
-        ts_iterator=test_ts_at_all_levels_it,
-        fcst_iterator=forecast_at_all_levels_unpacked_it,
+    num_levels = len(temporal_hierarchy.agg_multiples)
+    forecast_at_all_levels_unpacked_it_set = tee(
+        forecast_at_all_levels_unpacked_it,
+        num_levels,
     )
+    test_ts_at_all_levels_it_set = tee(test_ts_at_all_levels_it, num_levels)
+
+    def skip_iter(it, num_skips: int, offset: int):
```
Comment (melopeo) [inline src/gluonts/nursery/temporal_hierarchical_forecasting/eval/evaluation.py:112]: I am unsure if it is necessary to define this function at this place. Should we move it to the top of this file?

File: src/gluonts/nursery/temporal_hierarchical_forecasting/eval/evaluation.py [hunk 2]
```diff
@@ -87,27 +88,41 @@ def evaluate_forecasts_at_all_levels(
...
     metrics: List[str] = ["mean_wQuantileLoss"],
 ):
+    # (ts_1_6M, ts_1_2M, ts_1_1M, ts_2_6M, ts_2_2M, ts_2_1M, ...)
     forecast_at_all_levels_unpacked_it = unpack_forecasts(
         forecast_at_all_levels_it=forecast_at_all_levels_it,
         temporal_hierarchy=temporal_hierarchy,
         target_temporal_hierarchy=temporal_hierarchy,
     )
 
-    # First get item metrics for all time series for all frequencies; these are per time series metrics.
```
Comment (melopeo) [inline src/gluonts/nursery/temporal_hierarchical_forecasting/eval/evaluation.py:96]: I would suggest to keep a message like this one, as it explains why we are using `tee` in this particular way.

File: src/gluonts/nursery/temporal_hierarchical_forecasting/eval/evaluation.py (hunk 1 above)
Comment (rshyamsundar) [reply src/gluonts/nursery/temporal_hierarchical_forecasting/eval/evaluation.py:112]: It's a local utility function and is best kept there. I added comment on why we need it.

File: src/gluonts/nursery/temporal_hierarchical_forecasting/eval/evaluation.py (hunk 2 above)
Comment (rshyamsundar) [reply src/gluonts/nursery/temporal_hierarchical_forecasting/eval/evaluation.py:96]: Ya, added comments. Thanks!

**Review by melopeo — APPROVED** [review]
LGTM

**Pull request merged**